app/models/challenge_models/gift_exchange.rb

```diff
@@ -17,26 +17,28 @@ class GiftExchange < ApplicationRecord
   belongs_to :potential_match_settings, dependent: :destroy
   accepts_nested_attributes_for :potential_match_settings
 
-  validates_length_of :signup_instructions_general, :signup_instructions_requests, :signup_instructions_offers, {
+  validates :signup_instructions_general, :signup_instructions_requests, :signup_instructions_offers, length: {
     allow_blank: true,
     maximum: ArchiveConfig.INFO_MAX, too_long: ts("must be less than %{max} letters long.", max: ArchiveConfig.INFO_MAX)
   }
 
   PROMPT_TYPES.each do |type|
-    %w(required allowed).each do |setting|
+    %w[required allowed].each do |setting|
       prompt_limit_field = "#{type}_num_#{setting}"
-      validates_numericality_of prompt_limit_field, only_integer: true, less_than_or_equal_to: ArchiveConfig.PROMPTS_MAX, greater_than_or_equal_to: 0
+      validates prompt_limit_field, numericality: {
+        only_integer: true,
+        less_than_or_equal_to: ArchiveConfig.PROMPTS_MAX,
+        greater_than_or_equal_to: 1
+      }
     end
   end
 
   before_validation :update_allowed_values
   def update_allowed_values
-    %w(request offer).each do |prompt_type|
-      required = eval("#{prompt_type}s_num_required")
-      allowed = eval("#{prompt_type}s_num_allowed")
-      if required > allowed
-        eval("#{prompt_type}s_num_allowed = required")
-      end
+    %w[request offer].each do |prompt_type|
+      required = send("#{prompt_type}s_num_required")
+      allowed = send("#{prompt_type}s_num_allowed")
+      send("#{prompt_type}s_num_allowed=", required) if required > allowed
```

Comment:
So, the old code was assigning the _string_ `required`? And somehow nothing exploded?

Should similar code in [challenge_core.rb](https://github.com/otwcode/otwarchive/blob/317e3e840ce28f1d66e2ede43fdbd3e6bf0218bd/lib/challenge_core.rb#L7) and [prompt_restriction.rb](https://github.com/otwcode/otwarchive/blob/317e3e840ce28f1d66e2ede43fdbd3e6bf0218bd/app/models/prompt_restriction.rb#L33) be likewise updated? Though might be best to do so in a different ticket/PR.

Reply:
The old code just doesn't change the value. I _think_ the `eval` was just setting the value for its own context. The second example -- prompt restriction -- does work because it uses `self.<method>`. We could fix this one as well by doing
```rb
eval("self.{prompt_type}s_num_allowed = required")
```
but `eval` is unnecessarily dangerous when `send` will do. At some point we'll want to change the other `eval`s too. That first link -- challenge core -- does seem to have the same problem, so I'll poke a bit more and write something up.